src/main/java/com/rusticisoftware/tincan/RemoteLRS.java

```diff
@@ -504,6 +598,18 @@ public StatementsResultLRSResponse saveStatements(List<Statement> statements) {
             return lrsResponse;
         }
 
+        for (Statement statement: statements) {
+            if (statement.hasAttachmentsWithContent()) {
+                if(lrsResponse.getRequest().getPartList() == null){
+                    lrsResponse.getRequest().setPartList(statement.getPartList());
+                }
+                else{
+                    lrsResponse.getRequest().getPartList().addAll(statement.getPartList());
+                }
```

Comment:
Rather than doing this in if/else I think it'd be more clear to just instantiate a new list in the null case, then always `.addAll`, that also prevents there from being a reference to the first list in the request and having other attachments added to that list.